lib/customer.rb

```diff
@@ -0,0 +1,33 @@
+require 'csv'
+require 'ap'
+
+class Customer
+
+  attr_reader :id
+  attr_accessor :email, :address
+
+  def initialize(id, email, address)
+
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all
+
+
+    CSV.open("data/customers.csv", "r").map do |row|
+      address = row[2] + ' ' + row[3] + ' ' +  row[4] + ' ' +  row[5]
+      Customer.new(row[0].to_i, row[1], address)
+    end
```

Comment:
I find that this implicit return isn't quite as clear as I want-- I don't realize right away that `CSV.open(...)` is chained with `.map` and what that is doing. That being said... I realized it eventually lol